source/gravity_model/interface.cc

```diff
@@ -98,6 +98,10 @@ namespace aspect
       }
       prm.leave_subsection ();
 
+      AssertThrow(model_name != "",
```

Comment:
I can't see how you would even get here. The gravity model is declared with `Patterns::Selection`, so not getting a gravity model will yield an error already when the parameter file is read -- you won't even get here. I suspect that's the same for all of the other models you test in this patch.